examples/diff.c

```diff
@@ -335,3 +349,58 @@ static void diff_print_stats(git_diff *diff, struct opts *o)
 	git_buf_free(&b);
 	git_diff_stats_free(stats);
 }
+
+int string_to_int(const char * cmd_opt)
+{
+	int to_return = 0;
+	int i = 0;
+
+
+	while(cmd_opt[i]!='%') {
+		to_return *= 10;
+		to_return += cmd_opt[i] - '0';
+		i++;
+	}
+	return to_return;
+}
```

Comment:
This function doesn't make sure that the user did pass in a number. You could pass in "foo" and this would use the difference between `'f'` and `'0'` as the number. It also only terminates on `'%'` which means that if you do not pass in the percent, it'll run off the end of the string, causing a segfault.

I would recommend using a standard function like `strtol()` like we do elsewhere to parse strings into numbers. You can then check that what comes after the number is allowed in this context.